integration/spark/src/main/scala/org/apache/spark/sql/hive/CarbonPreInsertionCasts.scala

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
Done, currently to keep 1.5 and 2.1 in sync, CarbonAnalysisRules.scala will keep the rules of IUD and Insert Select. Created CarbonAnalysisRules.scala in 2.1 where currenly on Insert-Select is there as IUD is still not ported to 2.1